cfme/tests/cloud_infra_common/test_retirement.py

```diff
@@ -141,3 +142,31 @@ def test_unset_retirement_date(test_vm):
 
     test_vm.set_retirement_date(None)
     verify_retirement_date(test_vm, expected_date='Never')
+
+
+@test_requirements.retirement
+@pytest.mark.meta(blockers=[BZ(1306471, unblock=lambda provider: isinstance(provider,
+                                                                            InfraProvider))])
+@pytest.mark.parametrize('remove_date', [True, False], ids=['remove_date', 'set_future_date'])
+def test_resume_retired_instance(test_vm, provider, remove_date):
+    """Test resuming a retired instance, should be supported for infra and cloud, though the
+    actual recovery results may differ depending on state after retirement
+
+    Two methods to resume:
+    1. Set a retirement date in the future
+    2. Remove the set retirement date
+    """
+    num_days = 5
+
+    test_vm.retire()
+    verify_retirement_state(test_vm)
+
+    retire_date = None if remove_date else generate_retirement_date(delta=num_days)
+    test_vm.set_retirement_date(retire_date)
+
+    if BZ(1419150, forced_streams='5.6').blocks and not remove_date:
+        # The date is wrong, but we can still test unset
+        logger.warning('Skipping test step verification for BZ 1419150')
+    else:
+        verify_retirement_date(test_vm, expected_date=retire_date if retire_date else 'Never')
+    assert test_vm.is_retired is False
```

Comment:
I think I disagree with you here Ievgen, because although 'is' makes an identity comparison, I want to assert that the response from is_retired is boolean False, and doesn't equate to boolean false (ex. `0`).

Identity comparison is used throughout our codebase for booleans. Definitely not for non-boolean value comparison though.